awscrt/s3.py

```diff
@@ -0,0 +1,282 @@
+"""
+S3 client
+"""
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+import _awscrt
+from concurrent.futures import Future
+from awscrt import NativeResource
+from awscrt.http import HttpRequest
+from awscrt.io import ClientBootstrap, TlsConnectionOptions
+from awscrt.auth import AwsCredentialsProvider
+import awscrt.exceptions
+import threading
+from enum import IntEnum
+
+
+class S3RequestType(IntEnum):
+    """The type of the Aws S3 request"""
+
+    DEFAULT = 0
+    """
+    Default type, which is all the rest type of S3 requests besides of GET_OBJECT/PUT_OBJECT
+    """
+
+    GET_OBJECT = 1
+    """
+    Get Object S3 request
+    """
+
+    PUT_OBJECT = 2
+    """
+    Put Object S3 request
+    """
+
+
+class S3RequestTlsMode(IntEnum):
+    """Tls mode for S3 request"""
+
+    AWS_MR_TLS_ENABLED = 0
```

Comment:
what's MR?

Reply:
Copy/pasted from https://github.com/awslabs/aws-c-s3/blob/main/include/aws/s3/s3_client.h#L54-L57.
Should stand for meta_request. Hmm, since we are not taking meta request to python land, we probably should get out of it🤔